steps/rust-caching.yml

```diff
@@ -25,17 +25,16 @@ steps:
           Write-Host "##vso[task.setvariable variable=homedir]$env:HOME"
         displayName: Get Home directory (windows)
         condition: eq(variables['Agent.OS'], 'Windows_NT')
-  - ${{ if parameters.cacheRegistry }}:
       - task: Cache@2
         inputs:
-          key: '"rust-${{ parameters.projectName }}-cache" | "$(Agent.OS)" | "${{ parameters.workingDirectory }}/Cargo.toml"'
+          key: '"rust-${{ parameters.projectName }}-cache" | "$(Agent.OS)" | ${{ parameters.workingDirectory }}/Cargo.toml'
```

Comment:
If I'm reading the docs correctly, double-quoted key segments are treated as fixed names. Leaving off the quotes should treat it as a path, so the contents of the file get hashed and used as part of the key

https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops#key-input